management/src/main/java/org/ehcache/management/providers/statistics/StandardEhcacheStatistics.java

```diff
@@ -73,145 +79,100 @@
-          return new CounterHistory(buildHistory(count, since), NumberUnit.COUNT);
-        } else if ((name + "Rate").equals(statisticName)) {
-          SampledStatistic<Double> rate = compoundOperation.compound((Set) registeredCompoundStatistic.getCompound()).rate();
-          return new RateHistory(buildHistory(rate, since), TimeUnit.SECONDS);
-
-        } else if ((name + "LatencyMinimum").equals(statisticName)) {
-          SampledStatistic<Long> minimum = compoundOperation.compound((Set) registeredCompoundStatistic.getCompound()).latency().minimum();
-          return new DurationHistory(buildHistory(minimum, since), TimeUnit.NANOSECONDS);
-
-        } else if ((name + "LatencyMaximum").equals(statisticName)) {
-          SampledStatistic<Long> maximum = compoundOperation.compound((Set) registeredCompoundStatistic.getCompound()).latency().maximum();
-          return new DurationHistory(buildHistory(maximum, since), TimeUnit.NANOSECONDS);
-
-        } else if ((name + "LatencyAverage").equals(statisticName)) {
-          SampledStatistic<Double> average = compoundOperation.compound((Set) registeredCompoundStatistic.getCompound()).latency().average();
-          return new AverageHistory(buildHistory(average, since), TimeUnit.NANOSECONDS);
-        }
-      } else if (registeredStatistic instanceof RegisteredRatioStatistic) {
-        RegisteredRatioStatistic registeredRatioStatistic = (RegisteredRatioStatistic) registeredStatistic;
-        CompoundOperation<?> compoundOperation = registeredRatioStatistic.getCompoundOperation();
-
-        if (name.equals(statisticName)) {
-          SampledStatistic<Double> ratio = (SampledStatistic) compoundOperation.ratioOf((Set) registeredRatioStatistic.getNumerator(), (Set) registeredRatioStatistic.getDenominator());
-          return new RatioHistory(buildHistory(ratio, since), NumberUnit.RATIO);
-        }
-      } else if (registeredStatistic instanceof RegisteredSizeStatistic) {
-        RegisteredSizeStatistic registeredSizeStatistic = (RegisteredSizeStatistic) registeredStatistic;
-        if (name.equals(statisticName)) {
-          SampledStatistic<Long> count = (SampledStatistic<Long>) registeredSizeStatistic.getSampledStatistic();
-          return new SizeHistory(buildHistory(count, since), MemoryUnit.B);
-        }
-      } else if (registeredStatistic instanceof RegisteredCounterStatistic) {
-        RegisteredCounterStatistic registeredCounterStatistic = (RegisteredCounterStatistic) registeredStatistic;
-        if (name.equals(statisticName)) {
-          SampledStatistic<Long> count = (SampledStatistic<Long>) registeredCounterStatistic.getSampledStatistic();
-          return new CounterHistory(buildHistory(count, since), NumberUnit.COUNT);
-        }
-      } else {
-        throw new UnsupportedOperationException("Cannot handle registered statistic type : " + registeredStatistic);
+  Statistic<?, ?> queryStatistic(String statisticName, long since) {
```

Comment:
I've seen the exact same code in `AbstractExposedStatistics`. Could you refactor this code to extract the common bits of code?